File: shoring/blackjackEngine/src/main/java/com/panopset/blackjackEngine/BlackjackConfigDefault.java
package com.panopset.blackjackEngine;

import java.util.List;
import com.panopset.compat.Stringop;

public class BlackjackConfigDefault implements BlackjackConfiguration {

  public static final Integer DEFAULT_INITIAL_STAKE_IWD = 300;
  public static final Integer DEFAULT_MIN_BET_IWD = 5;
  public static final Integer DEFAULT_BET_INCR_IWD = 5;
  public static final Integer DEFAULT_LARGE_BET_IWD = 20;
  public static final Integer DEFAULT_TARGET_STAKE_IWD = 10000;
  public static final Integer DEFAULT_NBR_DECKS = 8;
  public static final Integer DEFAULT_NBR_SEATS = 1;
  private int seats = DEFAULT_NBR_SEATS;
  private int decks = DEFAULT_NBR_DECKS;
  private boolean isFastDeal = false;
  private boolean isVariations = false;

  public static final BlackjackConfigDefault INSTANCE = new BlackjackConfigDefault();

  private boolean isDASallowed = true;
  private boolean isRSAallowed = false;
  private boolean isDLRhitSoft17 = true;
  private boolean isEuropeanStyle = false;
  private boolean isLateSurrenderAllowed = false;
  private boolean isBlackjack6to5 = false;
  private boolean isEvenMoneyOnBlackjackVace = false;

  @Override
  public int getSeats() {
    return seats;
  }

  @Override
  public int getDecks() {
    return decks;
  }

  @Override
  public boolean isFastDeal() {
    return isFastDeal;
  }

  @Override
  public boolean isBasicStrategyVariationsOnly() {
    return isVariations;
  }

  @Override
  public boolean isDoubleAfterSplitAllowed() {
    return isDASallowed;
  }

  @Override
  public boolean isResplitAcesAllowed() {
    return isRSAallowed;
  }

  private List<String> sd;

  @Override
  public List<String> getStrategyData() {
    if (sd == null) {
      sd = Stringop.stringToList(DefaultResources.getDefaultBasicStrategy());
    }
    return sd;
  }

  private List<String> csd;
  
  @Override
  public List<String> getCountingSystemData() {
    if (csd == null) {
      csd = Stringop.stringToList(DefaultResources.getDefaultCountingSystems());
    }
    return csd;
  }

  @Override
  public boolean isDealerHitSoft17() {
    return isDLRhitSoft17;
  }

  @Override
  public boolean isCountVeryPositive() {
    return false;
  }

  @Override
  public boolean isEuropeanStyle() {
    return isEuropeanStyle;
  }

  @Override
  public boolean isCountVeryNegative() {
    return false;
  }

  @Override
  public String getStackedDeck() {
    return "";
  }

  @Override
  public boolean isLateSurrenderAllowed() {
    return isLateSurrenderAllowed;
  }
  
  @Override
  public boolean isBlackjack6to5() {
    return isBlackjack6to5;
  }

  @Override
  public boolean isEvenMoneyOnBlackjackVace() {
    return isEvenMoneyOnBlackjackVace;
  }
  
  private boolean showCount = true;
  
  @Override
  public boolean isShowCount() {
    return showCount;
  }

  @Override
  public void toggleShowCount() {
    showCount = !showCount;
  }

  @Override
  public int getLargeBetInWholeDollars() {
    return DEFAULT_LARGE_BET_IWD;
  }

  @Override
  public int getTargetStakeInWholeDollars() {
    return DEFAULT_TARGET_STAKE_IWD;
  }

  @Override
  public int getMinimumBetInWholeDollars() {
    return DEFAULT_MIN_BET_IWD;
  }

  @Override
  public int getBetIncrementInWholeDollars() {
    return DEFAULT_BET_INCR_IWD;
  }

  @Override
  public int getReloadAmountInWholeDollars() {
    return DEFAULT_INITIAL_STAKE_IWD;
  }

  BlackjackMessages messages;
  
  @Override
  public BlackjackMessages getMessages() {
    if (messages == null) {
      messages = new BlackjackMessagesDft();
    }
    return messages;
  }

  @Override
  public boolean isBetIdeaDoubleAfterBust() {
    return false;
  }

  @Override
  public boolean isBetIdeaLetItRideAfterTwoWins() {
    return false;
  }
}

File: code/shoring/blackjackEngine/src/main/kotlin/com/panopset/blackjackEngine/BlackjackConfigDefault.kt
package com.panopset.blackjackEngine

import com.panopset.compat.Stringop

class BlackjackConfigDefault: BlackjackConfiguration {
    var defaultResources = DefaultResources()
    private var sc = false
    override fun getLargeBetInWholeDollars(): Int {
     return 20
    }
    override fun getTargetStakeInWholeDollars(): Int {
        return 10000
    }
    override fun getMinimumBetInWholeDollars(): Int {
        return 5
    }
    override fun getBetIncrementInWholeDollars(): Int {
        return 5
    }
    override fun getReloadAmountInWholeDollars(): Int {
        return 300
    }
    override fun getSeats(): Int {
        return DEFAULT_NBR_SEATS
    }
    override fun getDecks(): Int {
        return DEFAULT_NBR_DECKS
    }
    override fun isFastDeal(): Boolean {
        return false
    }
    override fun isBasicStrategyVariationsOnly(): Boolean {
        return false
    }
    override fun isDoubleAfterSplitAllowed(): Boolean {
        return true
    }
    override fun isResplitAcesAllowed(): Boolean {
        return false
    }
    override fun isSplitAcePlayable(): Boolean {
        return false
    }
    override fun isDealerHitSoft17(): Boolean {
        return true
    }
    override fun isEuropeanStyle(): Boolean {
        return false
    }
    override fun isLateSurrenderAllowed(): Boolean {
        return false
    }
    override fun isBlackjack6to5(): Boolean {
        return false
    }
    override fun isEvenMoneyOnBlackjackVace(): Boolean {
        return false
    }
    override fun getStrategyData(): ArrayList<String> {
        return Stringop.stringToList(defaultResources.defaultBasicStrategy)
    }
    override fun getCountingSystems(): CountingSystems {
        return CountingSystems(Stringop.stringToList(defaultResources.defaultCountingSystems))
    }
    override fun getStrategicVeryPositiveCount(): Int {
        return 10
    }
    override fun isShowCount(): Boolean {
        return sc
    }

    override fun toggleShowCount() {
        sc = !sc
    }

    override fun getMessages(): BlackjackMessages {
        return BlackjackMessagesDft()
    }
}

const val DEFAULT_NBR_DECKS = 8
const val DEFAULT_NBR_SEATS = 1
